### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false negatives for nested declarations

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#7062]
### What minimal example or steps are needed to reproduce the bug?

```css
.foo {
	color: 10px; /* has a warning */
	color: red;
}

.foo {
	@media (screen) {
		color: 10px; /* no warning */
		color: red;
	}
}
```

_I added `color: red` to make sure there weren't any false positives._

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-property-value-no-unknown": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwA6A7bALpADYQBO8GAjAAwAOAHgNx6EnmVkxQu4C+rXKnRZWBAAIBbbgEsAhhgAUAZzBcYuAJSj8BNhFIVq9ZmP2HO3XoQH8QAGhAoZxGADk50hCBgNPdVyQwZWUHcAhcZwBzbxxcDAxsEDIAV1dlJMo4hISk2DBiOTI5AhkIgFo6Mgg6GDICAE9ygDc5YhSYctwIcpTcAGtugHd8BAwCVJg8BNs+MMhImSiAMXJJEu8AK2UIsNg6UMRsxJBlRtdiGVw9MaTCghgzpPtpk7OGi6uCcoXo8rO5LgoEUoJkTvdHjc8HM+EA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

I expected to see two errors.  
One for each `color: 10px;`.

### What actually happened?

Only one error was reported, for the regular, unnested rule.

**Comment by romainmenke** [comment]
https://github.com/stylelint/stylelint/blob/c812dae9da050f903d31f4786fff89929c56b0f6/lib/rules/declaration-property-value-no-unknown/index.js#L114

I think this was intended to differentiate at rule descriptors from declarations.

Maybe we can use the list of allowed nested at rules : https://drafts.csswg.org/css-nesting/#conditionals

> Specifically, these rules are capable of being [nested group rules](https://drafts.csswg.org/css-nesting/#nested-group-rules):
> all the [conditional group rules](https://drafts.csswg.org/css-conditional-3/#conditional-group-rule) ([@media](https://drafts.csswg.org/css-conditional-3/#at-ruledef-media), [@supports](https://drafts.csswg.org/css-conditional-3/#at-ruledef-supports))
> 
> - [@layer](https://drafts.csswg.org/css-cascade-5/#at-ruledef-layer)
> - [@scope](https://drafts.csswg.org/css-cascade-6/#at-ruledef-scope)
> - [@container](https://drafts.csswg.org/css-contain-3/#at-ruledef-container)
